src/Backend.cpp

```diff
@@ -68,6 +73,22 @@ string HABackend::CreateLongToken(string name)
 
 void HABackend::threadrunner()
 {
+  json getdomains;
+  getdomains["type"] = "get_services";
+  wc->send(getdomains);
+
+  auto msg = wc->recv();
+  json getdomainjson = json::parse(msg);
+  if (getdomainjson["id"] != getdomains["id"]) {
+    throw std::runtime_error("Didn't receive response to getDomains while we expected it");
```

Comment:
I wonder if this is entirely robust. HA might decide to send other things at random times - although we did not subscribe to anything yet. Fine for now at least!

Reply:
I agree, but this is why i throw that runtime error. If we ever get into a situation where this is not robust then we will know about it. (that's at least the thinking behind the throw at the moment)